src/controllers/queryRunner.ts

```diff
@@ -152,10 +152,48 @@ export default class QueryRunner {
     public async cancel(): Promise<QueryCancelResult> {
         // Make the request to cancel the query
         let cancelParams: QueryCancelParams = { ownerUri: this._ownerUri };
-        let queryCancelResult = await this._client.sendRequest(
-            QueryCancelRequest.type,
-            cancelParams,
+        let queryCancelResult: QueryCancelResult;
+        try {
+            queryCancelResult = await this._client.sendRequest(
+                QueryCancelRequest.type,
+                cancelParams,
+            );
+        } catch (error) {
+            // Ensure state is reset on error
+            this._isExecuting = false;
+            this._hasCompleted = true;
+            this.removeRunningQuery();
+            // Removed call to unregisterRunner (does not exist)
+            const promise = this._uriToQueryPromiseMap.get(this._ownerUri);
+            if (promise) {
+                promise.reject(error);
+                this._uriToQueryPromiseMap.delete(this._ownerUri);
+            }
+            this.eventEmitter.emit(
+                "complete",
+                Utils.parseNumAsTimeString(this._totalElapsedMilliseconds),
+                true,
+            );
+            this._statusView.executedQuery(this._ownerUri);
+            this._vscodeWrapper.showErrorMessage("Cancel failed: " + error.message);
+            return;
+        }
```

Comment:
Can we create a new method and move all the repeated code to that method?

Something like:

`completeExecution(error?: Error)`